### PR TITLE
Set number of holes

### PR DESCRIPTION
**Pull request by Hunt-E**
Added onto the startGame method by adding text to ask them for the amount of holes and get the input.

Removes the call to method nextTurn inside of setNumOfPlayers in the CLIView so that it doesn't start the game before setting the number of holes.

Created setNumOfHoles to set the number of holes in the controller and print out the amount of holes that have been selected.

Added setNumberOfHoles in the View and Controller so that it can be called inside of CLIView and be brought to Controller to process.

The number of holes can either be 9 or 18, and the method in Controller checks and throws an IllegalArgumentException if it is a different number.

Also added the private int numberOfHoles and numberOfPlayedHoles into Controller to keep track of them.

### CODE REVIEW

**Review by roym5 — APPROVED** [review]
looks good

**Review by roym5 — COMMENTED** [review]
Good methods in controller nice handling it not being 9/18

File: src/csc439teamplatypus/cardgame/golfgame/Controller.java [hunk 1]
```diff
@@ -230,6 +236,16 @@ protected void setNumberOfPlayers(int numberOfPlayers) {
         }
     }
 
+    protected void setNumberOfHoles(int numberOfHoles) {
```
Comment (abbiland) [inline src/csc439teamplatypus/cardgame/golfgame/Controller.java:239]: Looks great, thanks for updating the javadocs too!(:

**Comment by nflanagan6** [comment]
Hey @Hunt-E, sorry if I'm missing something, but are there hole methods in View/CLIView as well?

File: src/csc439teamplatypus/cardgame/golfgame/Controller.java (hunk 1 above)
Comment (benbaxter) [reply src/csc439teamplatypus/cardgame/golfgame/Controller.java:239]: I agree that the commit that changes setNumberOfHoles looks good. Clean, simple, to the point! The other commit adds a lot of extra "noise" to the spirit of the PR.  The java doc change should be its own PR which in itself is small, simple, and to the point. Now, looking through the history trying to understand the changes, it gets confusing why setNumberOfHole's PR changes so many other files and code that are unrelated.

I recommend checking out this small article on why small PRs are very useful and practical. 

https://google.github.io/eng-practices/review/developer/small-cls.html

**Pull request merged**